### PR TITLE
Structure tests by use case instead of one `happy_path` file

### PR DESCRIPTION
**Pull request by da-kami**
Moves tons of code around. Should be pure refactor.

### CODE REVIEW

File: daemon-tests/tests/none-collaborative_settlement.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
+use daemon::projection::CfdState;
```
Comment (klochowicz) [inline daemon-tests/tests/none-collaborative_settlement.rs]: nit: `non-collaborative-settlement` would be a much better name :P (I would also remove mix of hyphens and underscores, but I understand if you feel different).

**Review by klochowicz — APPROVED** [review]
a much needed TLC! :) thanks!

File: daemon-tests/tests/none-collaborative_settlement.rs (hunk 1 above)
Comment (luckysori) [reply daemon-tests/tests/none-collaborative_settlement.rs]: I think files (modules) have to respect the [naming rules](https://rust-lang.github.io/api-guidelines/naming.html), so snake_case it is.

**Comment by da-kami** [comment]
I just had a look at the test runtime for `macos` for different PRs.

For https://github.com/itchysats/itchysats/pull/2504 (the release) it took `18` minutes
For this PR it took `21` minutes (!)
For https://github.com/itchysats/itchysats/pull/2378 it took only `8` minutes (but that was already noted in https://github.com/itchysats/itchysats/issues/2473)

I wonder if splitting the tests into multiple files is not making them run longer, because test might not get triggered in parallel as flexible? Is there a correlation between tests and files, or are all tests from all files run in parallel?
Would this be relevant for https://github.com/itchysats/itchysats/issues/2473?

**Comment by da-kami** [comment]
> I just had a look at the test runtime for `macos` for different PRs.
> 
> For #2504 (the release) it took `18` minutes For this PR it took `21` minutes (!) For #2378 it took only `8` minutes (but that was already noted in #2473)
> 
> I wonder if splitting the tests into multiple files is not making them run longer, because test might not get triggered in parallel as flexible? Is there a correlation between tests and files, or are all tests from all files run in parallel? Would this be relevant for #2473?

If my understanding is correct, then *all* the tests should be run in parallel when we trigger `cargo test --workspace` on CI. So, putting the tests into separate files should not have any effect on test execution times. Please correct me if I'm wrong.

**Comment by klochowicz** [comment]
> > I just had a look at the test runtime for `macos` for different PRs.
> > For #2504 (the release) it took `18` minutes For this PR it took `21` minutes (!) For #2378 it took only `8` minutes (but that was already noted in #2473)
> > I wonder if splitting the tests into multiple files is not making them run longer, because test might not get triggered in parallel as flexible? Is there a correlation between tests and files, or are all tests from all files run in parallel? Would this be relevant for #2473?
> 
> If my understanding is correct, then _all_ the tests should be run in parallel when we trigger `cargo test --workspace` on CI. So, putting the tests into separate files should not have any effect on test execution times. Please correct me if I'm wrong.

it should not make much of a difference; I think this is definitely a step in the right direction. I will look into using https://nexte.st in our project to see the difference, it seems that it might help us a bit!

**Comment by da-kami** [comment]
CI ran into a failure where the `taker` dialer is unable to establish a connection, see: https://github.com/itchysats/itchysats/runs/7450380260?check_suite_focus=true

I was unable to reproduce this locally, will re-trigger the tests. This might point to some connection management problem.

**Comment by klochowicz** [comment]
> CI ran into a failure where the `taker` dialer is unable to establish a connection, see: https://github.com/itchysats/itchysats/runs/7450380260?check_suite_focus=true
> 
> I was unable to reproduce this locally, will re-trigger the tests. This might point to some connection management problem.

It's an issue that would fail the CI even with the legacy connection, although I have reported it as an issue as indeed it can be viewed as a regression that came with libp2p - in the old connection we'd exit this test soon after startup, instead of trying to pointlessly reconnect.
See the reported issue:
https://github.com/itchysats/itchysats/issues/2514

as this is an OS error, stating that we can't bind to the TCP socket, it should be reproducible to start maker config with dedicated libp2p port that is already used by something (even by another maker). I will write such test in the morning as a part of dealing with the bug noticed by you. I've added it to the iteration.

File: daemon-tests/tests/collaborative_settlement.rs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+async fn collaboratively_close_an_open_cfd_maker_going_short() {
+    collaboratively_close_an_open_cfd(Position::Short).await;
+}
+
+#[otel_test]
+async fn collaboratively_close_an_open_cfd_maker_going_long() {
+    collaboratively_close_an_open_cfd(Position::Long).await;
+}
+
+async fn collaboratively_close_an_open_cfd(maker_position: Position) {
```
Comment (luckysori) [inline daemon-tests/tests/collaborative_settlement.rs]: 🙃 I think it's better in test files when all the tests are at the top, and all the helpers at the bottom.

**Review by luckysori — APPROVED** [review]
Nice, thanks!

**Pull request merged**